cv32e40s/tests/programs/custom/cov_holes_generic/cov_holes_generic.c

```diff
@@ -506,6 +535,158 @@ int main(int argc, char **argv){
 
 // -----------------------------------------------------------------------------
 
+__attribute__((interrupt("machine")))
+void u_sw_irq_handler(void){
+  uint32_t  exccode;
+  uint32_t  instr_word;
+  uint32_t  mcause;
+  uint32_t  ret_addr;
+  uint32_t *mepc;
+
+
+  // Read CSRs
+
+  __asm__ volatile(
+    "csrr %[mcause], mcause"
+    : [mcause] "=r" (mcause)
+  );
+
+  exccode = mcause & 0xFFF;
+
+
+  // Handle causes
+
+  g_got_trap = 1;
+
+  if (exccode == EXC_CAUSE_ILLEGAL_INSTR) {
+    g_got_illegal_instruction_exception = 1;
+  }
+
+
+  // Setup mepc
+
+  __asm__ volatile(
+    "csrr %[mepc], mepc"
+    : [mepc] "=r" (mepc)
+  );
+
+  instr_word = *mepc;
+
+  if ((instr_word & 0x3) == 0x3) {
+    ret_addr = ((uint32_t)mepc) + 4;
+  } else {
+    ret_addr = ((uint32_t)mepc) + 2;
+  }
+
+  __asm__ volatile(
+    "csrw  mepc, %[ret_addr]"
+    : : [ret_addr] "r" (ret_addr)
+  );
+
+  return;
+}
+
+// -----------------------------------------------------------------------------
+
+void turn_on_dummies(void){
+  // "cpuctrl.rnddummy"
+  __asm__ volatile( "csrrs x0, 0xBF0, 2" );
+}
+
+// -----------------------------------------------------------------------------
+
+void test_lfsr_lockup(void){
```

Comment:
Looks good. 

Is it possible to detect if the minor alert gets set too? That would be interesting I think.

Since we are making a test for writing 0 to the secureseed registers, I think it would be nice if we also made a test for writing a "bad" seed to the secureseed registers. With "bad" seed I mean that eventually the LFSR would go into lockup by itself. This is expected behavior, and there is rtl code designed to make sure we handle this properly, so yeah it would be nice to test it too, if time allows.

Reply:
One moment...

Reply:
That is Perfect! 
Have you though about my comment about minor alert?

Reply:
Perfect!

Reply:
RESOLVED